hil-test/tests/lcd_cam_i8080.rs

```diff
@@ -328,12 +318,20 @@ mod tests {
             0b0001_0000_0000_0000,
         ];
 
-        static TX_BUF: ConstStaticCell<[u16; 10]> = ConstStaticCell::new([0; 10]);
-        let tx_buffer = TX_BUF.take();
-        tx_buffer.copy_from_slice(&data_to_send);
+        let mut dma_buf = ctx.dma_buf;
+
+        // FIXME: Replace this 16 -> 8 bit copy once DmaTxBuf takes a generic parameter.
```

Comment:
Imo we probably don't want DmaTxBuf and friends to be generic over the type, instead, it should always be a byte slice. If the slice fits the right requirements we can noop cast slices to other data types, i.e the slice has an align of 4 and length is a multiple of 4 we can create a u32 slice from that memory to fill the slice faster if needed.

Reply:
Why not `WriteBuffer`?  or are we deprecating this?  I found it handy to be able to implement `WriteBuffer` on a [frame buffer](https://github.com/liebman/esp-hub75/blob/main/src/framebuffer.rs#L199) and pass it in for DMA. (specifically the i8080 peripheral)

Reply:
> If the slice fits the right requirements

I suppose this is the main issue. Ideally users should have some kind of guarantee. 

I'm not too fussed about how this is achieved.

Reply:
We can hash this out in another PR/issue. @MabezDev would you prefer if I removed the comment?

Reply:
The comment is fine, we'll figure this out later.